chain/chain/src/chain.rs

```diff
@@ -4020,22 +4005,18 @@ impl<'a> ChainUpdate<'a> {
             if header.next_bp_hash() != prev_header.next_bp_hash() {
                 return Err(ErrorKind::InvalidNextBPHash.into());
             }
-        } else {
-            if header.next_bp_hash()
-                != &Chain::compute_bp_hash(
-                    &*self.runtime_adapter,
-                    header.next_epoch_id().clone(),
-                    header.epoch_id().clone(),
-                    &header.prev_hash(),
-                )?
-            {
-                return Err(ErrorKind::InvalidNextBPHash.into());
-            }
+        } else if header.next_bp_hash()
+            != &Chain::compute_bp_hash(
+                &*self.runtime_adapter,
+                header.next_epoch_id().clone(),
+                header.epoch_id().clone(),
+                header.prev_hash(),
+            )?
```

Comment:
It feels like the rhs here wants to be extracted in a name variable. It's a complex, multiline *and failable* expression, makes sense to pull it into a separate statment.